examples/all-clusters-app/all-clusters-common/include/WhmDelegate.h

```diff
@@ -0,0 +1,270 @@
+     *                      the tank SHALL be allowed to fall before again beginning to reheat it.
+     *
+     * @return  Success if the boost command is accepted; otherwise the command SHALL be rejected with appropriate error.
+     */
+    Protocols::InteractionModel::Status HandleBoost(uint32_t duration, Optional<bool> oneShot, Optional<bool> emergencyBoost,
+                                                    Optional<int16_t> temporarySetpoint, Optional<chip::Percent> targetPercentage,
+                                                    Optional<chip::Percent> targetReheat) override;
+
+    /**
+     * @brief Delegate should implement a handler to cancel a boost command.
+     *        Upon receipt, the Water Heater SHALL transition back from the BOOST state to the previous mode
+     *        (e.g. OFF, MANUAL or TIMED).
+     *
+     * @return It should report SUCCESS if successful and FAILURE otherwise.
+     */
+    Protocols::InteractionModel::Status HandleCancelBoost() override;
+
+    // ------------------------------------------------------------------
+    // Get attribute methods
+    BitMask<WaterHeaterTypeBitmap> GetHeaterTypes() override;
+    BitMask<WaterHeaterDemandBitmap> GetHeatDemand() override;
+    uint16_t GetTankVolume() override;
+    int64_t GetEstimatedHeatRequired() override;
+    Percent GetTankPercentage() override;
+    BoostStateEnum GetBoostState() override;
+
+    // ------------------------------------------------------------------
+    // Set attribute methods
+    void SetHeaterTypes(BitMask<WaterHeaterTypeBitmap> heaterTypes);
+    void SetHeatDemand(BitMask<WaterHeaterDemandBitmap> heatDemand);
+    void SetTankVolume(uint16_t tankVolume);
+    void SetEstimatedHeatRequired(int64_t estimatedHeatRequired);
+    void SetTankPercentage(Percent tankPercentage);
+    void SetBoostState(BoostStateEnum boostState);
+
+    /*********************************************************************************
+     *
+     * WaterHeaterManagementDelegate specific methods
+     *
+     *********************************************************************************/
+
+    /**
+     * @brief Set the Water Header Mode and act accordingly.
+     *
+     * @param mode  The Water Heater Mode (e.g. OFF, MANUAL or TIMED).
+     */
+    Protocols::InteractionModel::Status SetWaterHeaterMode(uint8_t mode);
+
+    /**
+     * @brief Set the water temperature of the tank
+     *
+     * @param waterTemperature  The water temperature in 100th's Celsius
+     */
+    void SetWaterTemperature(uint16_t waterTemperature);
+
+    /**
+     * @brief Set the target water temperature of the tank
+     *
+     * @param targetWaterTemperature  The water temperature in 100th's Celsius
+     */
+    void SetTargetWaterTemperature(uint16_t targetWaterTemperature);
+
+    /**
+     * @brief Determine whether the heating sources need to be turned on or off
```

Comment:
What does the status here mean?  Does success mean "on", "off", or "check succeeded and we decided something but won't tell you what"?

Reply:
Have added this to the follow up issue to track - https://github.com/project-chip/connectedhomeip/issues/34565